util/gpool/spmc/worker.go

```diff
@@ -58,14 +58,23 @@ func (w *goWorker[T, U, C, CT, TF]) run() {
 			if f == nil {
 				return
 			}
+			if f.GetStatus() == pooltask.PendingTask {
+				f.SetStatus(pooltask.RunningTask)
+			}
 			w.pool.subWaitingTask()
 			ctx := f.GetContextFunc().GetContext()
 			if f.GetResultCh() != nil {
 				for t := range f.GetTaskCh() {
+					if f.GetStatus() == pooltask.StopTask {
+						f.Done()
+						continue
+					}
 					f.GetResultCh() <- w.pool.consumerFunc(t.Task, f.ConstArgs(), ctx)
 					f.Done()
 				}
+				w.pool.ExitSubTask(f.TaskID())
 			}
+			f.Finish()
```

Comment:
Could we remove this function or remove `Done`? It seems that `f.Done` and `f.Finish()` do the same thing: `t.wg.Done()`

Reply:
In fact, They are the same thing. But if Finish is also called, here are two ```Done```. it is weird. ```Finish``` is to tell system this ```TaskBox``` has finished job.

Reply:
> Could we remove this function or remove `Done`?

If we remove it, we will block when to exit.